src/qt/pivx/coldstakingmodel.cpp

```diff
@@ -137,20 +142,43 @@ QVariant ColdStakingModel::data(const QModelIndex &index, int role) const
 
 bool ColdStakingModel::whitelist(const QModelIndex& modelIndex) {
     QString address = modelIndex.data(Qt::DisplayRole).toString();
+    if (addressTableModel->isWhitelisted(address.toStdString())) {
+        // address already in the whitelist
+        return false;
+    }
+
     int idx = modelIndex.row();
     beginRemoveRows(QModelIndex(), idx, idx);
```

Comment:
`beginRemoveRows` needs to be inside the block if `endRemoveRows ` is inside, otherwise you are calling begin without end.